tests/integration/test_using_cached_dependencies.py

```diff
@@ -394,6 +403,10 @@ def update_main_repo(env_data, repo_dir, tmpdir, new_dep_commits, dep_repo):
             if go_dep.endswith(".git"):
                 go_dep = go_dep[: -len(".git")]
             f.write(f"require {go_dep} {dep_version} \n")
+        if private:
+            result_tidy = subprocess.run(["go", "mod", "tidy"], cwd=repo_dir)
+            assert result_tidy.returncode != 0, f"The command failed with: {result_tidy.stderr}"
```

Comment:
Actually, different idea: instead of `go mod tidy`, run `go get {dep_name}@{dep_version}`, which updates the checksum and doesn't remove unnecessary modules